buildkite/benchmark/run.py

```diff
@@ -267,17 +351,141 @@ def execute_command(self, command, path=".", exit_on_failure=True, log_stdout=Tr
+        Run.print_env_vars()
+        # NOTE: `bm.command` is the raw arrowbench name; `bm.name` is f"arrowbench/{bm.command}"
+        # to disambiguate from labs/benchmarks versions when filtering.
+        bm_names = [bm.command for bm in benchmark_groups]
+        r_command = f"""
+        arrowbench::install_pipx();
+        arrowbench::install_benchconnect();
+
+        bm_df <- arrowbench::get_package_benchmarks();
+        arrowbench::run(
+            bm_df[bm_df$name %in% c({str(bm_names)[1:-1]}), ],
+            n_iter = 3L,
+            drop_caches = TRUE,
+            publish = TRUE,
+            run_id = '{os.getenv('RUN_ID')}',
+            run_name = '{os.getenv('RUN_NAME')}',
+            run_reason = '{os.getenv('RUN_REASON')}'
+        )
+        """
+
+        (_, tmp) = tempfile.mkstemp(suffix=".R")
+        tmp = Path(tmp)
+        with open(tmp, "w") as f:
+            f.write(r_command)
+
+        self.executor.execute_command(
+            f"R -f {tmp}",
+            path=self.root,
+            exit_on_failure=False,
+        )
+
+        # leave tempfile for validation if mock run
+        if not all([bg.mock_run for bg in benchmark_groups]):
+            tmp.unlink()
+
+
+class AdapterBenchmarkGroupsRunner(ConbenchBenchmarkGroupsRunner):
+    """
+    Runner class for benchadapt adapters
+
+    Presently an alias, as relevant differences are all handled via the
+    command.
+    """
+
+
+class Run:
+    def __init__(self, repo_params):
+        self.repo = repo_params["repo"]
+        self.root = repo_params["root"]
+        self.branch = repo_params["branch"]
+        self.setup_commands = repo_params["setup_commands"]
+        self.path_to_benchmark_groups_list_json = repo_params[
+            "path_to_benchmark_groups_list_json"
+        ]
+        self.url_for_benchmark_groups_list_json = repo_params[
+            "url_for_benchmark_groups_list_json"
+        ]
+        self.benchmarkable_type = os.getenv("BENCHMARKABLE_TYPE")
+        self.filters = json.loads(os.getenv("FILTERS", "{}"))
+        self.setup_commands_for_lang_benchmarks = repo_params[
+            "setup_commands_for_lang_benchmarks"
+        ]
+        self.env_vars = repo_params["env_vars"]
+        self.benchmark_groups = []
+        self.executor = CommandExecutor()
+
+    def capture_context(self):
+        post_logs_to_arrow_bci(f"runs/{run_id}", run_context())
+
     def setup_benchmarks_repo(self):
         # if benchmarks are located in the same repo as was cloned in utils.sh, do not clone it again
         if os.getenv("REPO") == self.repo:
             return
 
-        self.execute_command(f"git clone {self.repo}")
-        self.execute_command(f"git fetch && git checkout {self.branch}", self.root)
+        if not Path(self.root).exists():
+            self.executor.execute_command(f"git clone {self.repo}")
+
+        self.executor.execute_command(
+            f"git fetch && git checkout {self.branch}", self.root
+        )
```

Comment:
Made cloning conditional, because it was failing when labs/benchmarks and arrowbench both cloned arrowbench, specifically by complaining the directory already exists. We could `rm -rf` it and re-clone, but given it then fetches anyway, I'm pretty sure it'd be duplicative.